server/src/microservices/processors/metadata-extraction.processor.ts

```diff
@@ -178,165 +111,117 @@ export class MetadataExtractionProcessor {
+    if (!this.reverseGeocodingEnabled || !longitude || !latitude) {
+      return exifData;
     }
-  }
 
-  private extractDuration(duration: number | string | null) {
-    const videoDurationInSecond = Number(duration);
-    if (!videoDurationInSecond) {
-      return null;
+    try {
+      const geocode = await this.geocodingRepository.reverseGeocode({ latitude, longitude });
+      return { ...exifData, ...geocode };
+    } catch (error: any) {
+      this.logger.warn(
+        `Unable to run reverse geocoding due to ${error} for asset ${asset.id} at ${asset.originalPath}`,
+        error?.stack,
+      );
     }
+    return exifData;
+  }
+
+  private async exifData(asset: AssetEntity): Promise<[ExifEntity, ImmichTags]> {
+    const readTaskOptions = {
+      defaultVideosToUTC: false,
+      useMWG: true,
+      numericTags: ['Duration', 'FocalLength'],
+      includeImageDataMD5: false,
+      geoTz: (lat: number, lon: number): string => geotz.find(lat, lon)[0],
+      optionalArgs: [],
+    };
+
+    const mediaTags = await exiftool
+      .read<ImmichTags>(asset.originalPath, undefined, readTaskOptions)
+      .catch((error: any) => {
+        this.logger.warn(`error reading exif data (${asset.id} at ${asset.originalPath}): ${error}`, error?.stack);
+        return null;
+      });
 
-    return Duration.fromObject({ seconds: videoDurationInSecond }).toFormat('hh:mm:ss.SSS');
+    const sidecarTags = asset.sidecarPath
+      ? await exiftool.read<ImmichTags>(asset.sidecarPath, undefined, readTaskOptions).catch((error: any) => {
+        this.logger.warn(`error reading exif data (${asset.id} at ${asset.sidecarPath}): ${error}`, error?.stack);
+        return null;
+      })
+      : null;
+
+    const stats = fs.statSync(asset.originalPath);
+
+    const tags = { ...mediaTags, ...sidecarTags };
+
+    const exifDate = (dt: ExifDateTime | string | undefined) => (dt instanceof ExifDateTime ? dt?.toDate() : null);
+
+    const validate = <T>(value: T): T | null => (typeof value === 'string' ? null : value ?? null);
+
+    return [
+      <ExifEntity>{
+        // altitude: tags.GPSAltitude ?? null,
+        assetId: asset.id,
+        dateTimeOriginal: exifDate(tags.DateTimeOriginal) ?? asset.fileCreatedAt,
+        exifImageHeight: validate(tags.ImageHeight),
+        exifImageWidth: validate(tags.ImageWidth),
+        exposureTime: tags.ExposureTime ?? null,
+        fileSizeInByte: stats.size,
+        fNumber: validate(tags.FNumber),
+        focalLength: validate(tags.FocalLength),
+        fps: validate(tags.VideoFrameRate),
+        iso: validate(tags.ISO),
+        latitude: validate(tags.GPSLatitude),
+        lensModel: tags.LensModel ?? null,
+        longitude: validate(tags.GPSLongitude),
+        make: tags.Make ?? null,
+        model: tags.Model ?? null,
+        modifyDate: exifDate(tags.ModifyDate) ?? asset.fileModifiedAt,
+        orientation: validate(tags.Orientation)?.toString() ?? null,
```

Comment:
This is fine for images, but videos are using degrees. There will need to be some sort of migration at some point. I believe clients look for both enums and degrees which makes things easier, but records in the db will still need to be corrected.

This also shouldn't be a string really.

Reply:
Yes, I am OK dealing with this one in the future. It looks like you usually will have values of 1-8 and we should convert the "degrees" variants to probably use that standard in the future.